src/evict/evict_lru.c

```diff
@@ -1355,7 +1353,7 @@ __evict_walk_file(WT_SESSION_IMPL *session,
 		 * no good eviction candidates can be found.  Abandon the walk
 		 * if we get into that situation.
 		 */
-		give_up = pages_seen > 100 &&
+		give_up = !__wt_cache_aggressive(session) && pages_seen > 100 &&
```

Comment:
@agorrod I didn't call this out but we discussed it.  I can move this part into a separate branch if you like (e.g., for WT-2891)?